src/AWS/Lambda/RuntimeClient.hs

```diff
@@ -112,9 +112,16 @@ runtimeClientRetryTry' :: Int -> IO (Response a) -> IO (Either HttpException (Re
 runtimeClientRetryTry' 1 f = try f
 runtimeClientRetryTry' i f = do
   resOrEx <- try f
+  let retry = threadDelay 500 >> runtimeClientRetryTry' (i - 1) f
   case resOrEx of
-    Left (_ :: HttpException) -> threadDelay 500 >> runtimeClientRetryTry' (i - 1) f
-    Right res -> return $ Right res
+    Left (_ :: HttpException) -> retry
+    Right res ->
+      -- TODO: Explore this further.
+      -- Before ~July 22nd 2020 it seemed that if a next event request reached
+      -- the runtime before a new event was available that there would be a
+      -- network error.  After it appears that a 403 is returned.
+      if getResponseStatus res == status403 then retry
```

Comment:
I'm curious, how it is going to help

Reply:
This comment goes over it pretty well:  https://github.com/Nike-Inc/hal/issues/54#issuecomment-662838349 and I just had an idea on how better to validate this (I can't be sure on what it _used_ to do, but what it does now).

Essentially the ideal order for lambda is:
  - The Container wakes
  - The AWS runtime sidecar has a new event available
  - The Haskell runtime GETs that event from the sidecar
  - The Haskell runtime POSTs the result to the sidecar based on user code
  - The Container sleeps
  - Repeat

But in reality, the GET happens before the container sleeps, because it's nearly immediate.  And there's no sense waiting, because the sleep is going to happen anyway, and there's no way to know when the sleep will occur or if it has.  The AWS docs say to ensure that there is no timeout, because it's possible this sleep is for a long time.  So the essentially expect that a GET for one invocation may have started in the previous.

Occasionally, we would see network errors, so this retry was added, and it's effects were apparent and observable.  There were lots of reasons this could happen, so I didn't think too much about it at the time, it just seemed important for robustness.

My theory now is that a change went in that prevented the network errors, but now return a 403 if the GET arrived _just before_ the next event was available, but after the POST result was processed.

Reply:
👍